cluster-autoscaler/FAQ.md

```diff
@@ -297,140 +308,143 @@ kubectl annotate node <nodename> cluster-autoscaler.kubernetes.io/scale-down-dis
-do it anymore as there is almost no capacity left. It has to them somewhere else, and it is not sure that
-if A had been deleted much earlier than B, during the last 10 min, would always have a place to
-move its pods. So the requirement of being unused for 10 min may not be valid anymore for B.
-But if another node C, in case of deletion, can move its pods to node Y then it
-may still do it, because no one touched Y. So C can be deleted immediately after A, but B may not be
-deleted immediately.
+different strategies can be selected for choosing which node group is increased. Check [What are Expanders?](#what-are-expanders) section to learn more about strategies.
+
+It may take some time before the created nodes appear in Kubernetes. It almost entirely
+depends on the cloud provider and the speed of node provisioning. Cluster
+Autoscaler expects requested nodes to appear within 15 minutes
+(configured by `--max-node-provision-time` flag.) After this time, if they don't
+appear, it stops considering them in simulations and may attempt to scale up a
+different group if the pods are still pending.
+
+### How does scale-down work?
+
+Every 10 seconds (configurable by `--scan-interval` flag), if no scale-up is
+needed, Cluster Autoscaler checks which nodes are unneeded. A node is considered for removal when:
+
+* The sum of cpu and memory requests of all pods running on this node is smaller
+  than 50% of the node's capacity. Utilization threshold can be configured using
+  `--scale-down-utilization-threshold` flag.
+
+* All pods running on the node (except these that run on all nodes by default, like manifest-run pods
+or pods created by daemonsets) can be moved to other nodes. Stand-alone pods which are not
+under control of a deployment, replica set, replication controller or job, would not be recreated
+if the node was deleted, so they can't be safely moved and the node can't be removed,
+even if its utilization is low. While checking this condition, the new locations of all moved pods
+are memorized. With that, Cluster Autoscaler knows where each pod can be moved, and which nodes
+depend on which other nodes in terms of pod migration. Of course, it may happen that eventually
+the scheduler will place the pods somewhere else.
+
+* All system pods running on the node (except these that run on all nodes by default, like
+manifest-run pods or pods created by daemonsets) have a PodDisruptionBudget
+which allows for evicting all of the replicas present on that node.
+
+* There are no pods with local storage (EmptyDir or HostPath volumes.) Applications with local storage would lose their data if a node was deleted, even if they are replicated.
+
+* It doesn't have scale-down disabled annotation (see [How can I prevent Cluster Autoscaler from scaling down a particular node?](#how-can-i-prevent-cluster-autoscaler-from-scaling-down-a-particular-node))
+
+If a node is unneeded for more than 10 minutes (configurable by `--scale-down-unneeded-time` flag)
```

Comment:
That is not really true. There are a lot more flags that impact how quickly CA deletes nodes than just `--scale-down-unneeded-time` (namely `--scale-down-delay-after-add`, `--scale-down-delay-after-delete`, `--scale-down-delay-after-failure`, which BTW have been renamed in 1.0). I have seen users change just `--scale-down-unneeded-time` and contact us after CA was still not deleting their nodes as quickly as they thought. Let's not propagate this misconception and either don't go into details at all, or really describe how to change that.

Reply:
Actually it's even more complex - since 1.0 you get an additional 5 minutes delay from `UnremovableNodeRecheckTimeout`, which you can't control in anyway (5 minutes is hardcoded). So effectively you get 5 minutes latency before it's even detected that node is unneeded.

Reply:
Replaced this with the reference to a section where all those flags are mentioned.